server/events/working_dir.go

```diff
@@ -89,17 +89,41 @@ func (w *FileWorkspace) Clone(
 		}
 		revParseCmd := exec.Command("git", "rev-parse", pullHead) // #nosec
 		revParseCmd.Dir = cloneDir
-		output, err := revParseCmd.CombinedOutput()
-		if err != nil {
-			log.Warn("will re-clone repo, could not determine if was at correct commit: %s: %s: %s", strings.Join(revParseCmd.Args, " "), err, string(output))
+		outputRevParseCmd, errRevParseCmd := revParseCmd.CombinedOutput()
+		if errRevParseCmd != nil {
+			log.Warn("will re-clone repo, could not determine if was at correct commit: %s: %s: %s", strings.Join(revParseCmd.Args, " "), err, string(outputRevParseCmd))
 			return w.forceClone(log, cloneDir, headRepo, p)
 		}
-		currCommit := strings.Trim(string(output), "\n")
+		currCommit := strings.Trim(string(outputRevParseCmd), "\n")
+
+		// We're bring our remote refs up to date
+		remoteUpdateCmd := exec.Command("git", "remote", "update")
+		remoteUpdateCmd.Dir = cloneDir
+		outputRemoteUpdate, errRemoteUpdate := remoteUpdateCmd.CombinedOutput()
+		if errRemoteUpdate != nil {
+			log.Warn("getting remote update failed: %s", string(outputRemoteUpdate))
+		}
+
+		// We're checking if remote master branch has diverged
+		statusUnoCmd := exec.Command("git", "status", "-uno")
```

Comment:
```suggestion
		statusUnoCmd := exec.Command("git", "status", "--untracked-files=no")
```
Will make it clearer what the flag will do

Reply:
Good idea, it is clearer!